src/editors/containers/ProblemEditor/components/EditProblemView/hooks.js

```diff
@@ -68,17 +68,27 @@ export const fetchEditorContent = ({ format }) => {
 export const parseState = ({
   problem,
   isAdvanced,
+  isMarkdownEditorEnabled,
   ref,
   lmsEndpointUrl,
 }) => () => {
-  const rawOLX = ref?.current?.state.doc.toString();
-  const editorObject = fetchEditorContent({ format: '' });
-  const reactOLXParser = new ReactStateOLXParser({ problem, editorObject });
+  const contentString = ref?.current?.state.doc.toString();
+  const rawOLX = isMarkdownEditorEnabled ? convertMarkdownToXml(contentString) : contentString;
+  let reactBuiltOlx;
+  if (!isMarkdownEditorEnabled) {
+    const editorObject = fetchEditorContent({ format: '' });
+    const reactOLXParser = new ReactStateOLXParser({ problem, editorObject });
+    reactBuiltOlx = setAssetToStaticUrl({ editorValue: reactOLXParser.buildOLX(), lmsEndpointUrl });
+  }
   const reactSettingsParser = new ReactStateSettingsParser({ problem, rawOLX });
-  const reactBuiltOlx = setAssetToStaticUrl({ editorValue: reactOLXParser.buildOLX(), lmsEndpointUrl });
+  const settings = isAdvanced ? reactSettingsParser.parseRawOlxSettings() : reactSettingsParser.getSettings();
   return {
-    settings: isAdvanced ? reactSettingsParser.parseRawOlxSettings() : reactSettingsParser.getSettings(),
-    olx: isAdvanced ? rawOLX : reactBuiltOlx,
+    settings: {
+      ...settings,
+      ...(isMarkdownEditorEnabled && { markdown: contentString }),
+      markdown_edited: isMarkdownEditorEnabled,
+    },
+    olx: isAdvanced || isMarkdownEditorEnabled ? rawOLX : reactBuiltOlx,
```

Comment:
This code seems fine to me but it takes a while to understand and figure out the different scenarios. Do you think you could add a couple lines of comments here just explaining the three different modes (advanced, markdown, and visual), and how that affects the OLX parsing and settings parsing? Basically summarizing the logic that's below.